### ISSUE TITLE
Issues with M1 ARM Macs

### ISSUE THREAD

**Issue by bdoherty-tovala** [fergusstrange/embedded-postgres#86]
Using version 19.0 which downloads `embedded-postgres` 14.5.0.

Currently the [`defaultVersionStrategy`](https://github.com/fergusstrange/embedded-postgres/blob/3f49b0df2af78200e3cc4cc511ed3f0876b304e6/version_strategy.go#L40) overrides detected ARM architectures to use the `amd64` architecture download for macs instead.

However this version seems to have some issues when running certain commands: in my case, a `create function` would always hang. This seems to be a problem specific to the amd binary running on the arm architecture.

For postgres versions 10.20, 11.15, 12.10, 13.6 and then >= 14.2, however, native ARM builds DO exist: https://repo1.maven.org/maven2/io/zonky/test/postgres/embedded-postgres-binaries-darwin-arm64v8/

I found that builds native to the ARM/M1 architecture don't exhibit the same problem: the function is created as expected and the process does not hang.

If the exception in the `defaultVersionStrategy` for versions >= 14.2 were removed, allowing the `darwin/arm64v8` binaries to be used on these architectures, it would unblock users that run M1 macs that are experiencing this problem.

**Comment by fergusstrange** [comment]
This is a fantastic fix @bdoherty-tovala. Can you confirm your forked code has been working well for the last few weeks?

**Comment by bdoherty-tovala** [comment]
Been working for me on an M1 mac and my colleagues on their Intel Macs. Hasn't broken anyone as far as I can tell :)

**Comment by fergusstrange** [comment]
This is fantastic. I'll look to get those changes merged into master soon then. I'll leave this open to reference the work when done! Cheers

**Comment by dcaf-mocha** [comment]
Hey @bdoherty-tovala curious how you got your fork to work I've tried `replace github.com/fergusstrange/embedded-postgres => github.com/tovala/embedded-postgres v0.0.0-20221103205757-5f8ea8a8920d` in my `go.mod`, but looking at my deps, I cant seem to see your changes in `version_strategy.go`

![image](https://user-images.githubusercontent.com/93599851/201489250-a0667695-1e83-491e-ad32-91c1c56379e7.png)

**Comment by bdoherty-tovala** [comment]
@fergusstrange is this working for you yet?

**Comment by fergusstrange** [comment]
Hey @bdoherty-tovala I've cut the release at [v1.20.0](https://github.com/fergusstrange/embedded-postgres/releases/tag/v1.20.0) ready for your testing.
